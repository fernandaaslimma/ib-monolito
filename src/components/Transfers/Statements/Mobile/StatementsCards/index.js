import StatementsCards from "./StatementsCards";

export default StatementsCards;
