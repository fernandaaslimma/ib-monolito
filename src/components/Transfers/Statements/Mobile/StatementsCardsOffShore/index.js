import StatementsCardsOffShore from "./StatementsCardsOffShore";

export default StatementsCardsOffShore;
