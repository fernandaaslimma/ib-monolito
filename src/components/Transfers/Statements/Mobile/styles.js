import styled, { css } from "styled-components";
import { rem, remFontSize } from "../../../../styles/tools";
import {
  lighestgrey,
  gray300,
  gray200,
  negative300,
  darkGreen
} from "../../../../styles/settings";

export const FormHeader = styled.div`
  display: flex;
  flex-direction: column;
  overflow: hidden;
  height: auto;
`;

export const FormContent = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

export const AccTitle = styled.p`
  color: ${gray300};
  font-family: Lato Bold;
  font-size: ${remFontSize(16)};
  padding: ${rem(32)} ${rem(10)} ${rem(15)} ${rem(10)};

  ${({ center }) =>
    center &&
    css`
      text-align: center;
    `}
`;

export const Separator = styled.span`
  display: block;
  height: ${rem(1)};
  background: ${lighestgrey};
`;

export const Space = styled.span`
  display: block;
  padding: 0 ${rem(8)};
`;

export const RadioWrapper = styled.div`
  display: flex;
  flex-direction: column;
  padding: ${rem(8)} ${rem(8)} 0 ${rem(8)};
`;

export const AccountWrapper = styled.div`
  display: flex;
  flex-direction: column;
  border-radius: 8px;
  background-color: white;
  margin-top: ${rem(27)};
  margin-left: ${rem(16)};
  margin-right: ${rem(16)};
  height: ${rem(60)};

  @media (min-width: 600px) {
    margin-left: 0;
    margin-right: 0;
  }
`;

export const BalanceWrapper = styled(AccountWrapper)`
  margin-top: ${rem(4)};
`;

export const FutureTransactionsWrapper = styled(AccountWrapper)`
  margin-top: 0px;
  height: ${rem(49)};
  margin-bottom: ${rem(24)};
`;

export const AccountBalance = styled.div`
  display: flex;
  color: ${darkGreen};
  flex-direction: row;
  font-family: Lato;
  font-style: normal;
  font-weight: 800;
  font-size: ${remFontSize(20)};
  line-height: normal;
  text-align: left;
  letter-spacing: 0.667px;
  width: 100%;
  padding: ${rem(16)};
`;

export const Account = styled.div`
  display: flex;
  font-family: Lato;
  font-size: 14px;
  font-style: normal;
  color: ${gray200};
  line-height: normal;
  text-align: left;
  align-items: center;
  width: 100%;
  padding: ${rem(16)};
  flex-direction: row;
  font-weight: 700;
  letter-spacing: 0.667px;
  color: ${darkGreen};
  height: 100%;
`;

export const RadioAccount = styled.p`
  font-family: Lato;
  font-size: 16px;
  font-weight: 600;
  color: ${gray300};
  line-height: 125%;
  text-align: left;
  margin: 0;
`;

export const FilterWrapper = styled.div`
  display: flex;
  flex-direction: column;
  padding: ${rem(24)} 0 ${rem(32)} 0;
`;

export const FilterLabels = styled.div`
  font-family: Lato;
  font-style: normal;
  font-weight: bold;
  font-size: ${rem(16)};
  line-height: ${rem(19)};
  display: flex;
  align-items: center;
  text-align: center;
  padding: 0 ${rem(16)} 0 ${rem(16)};

  color: ${gray300};
`;

export const RadioButtonsWrapper = styled.div`
  display: flex;
  padding: ${rem(16)} ${rem(16)} ${rem(32)} ${rem(16)};
  overflow-x: auto;

  label + label {
    margin-left: ${rem(12)};
  }
`;

export const InputsWrapper = styled.div`
  display: flex;
  flex-direction: row;
  padding: ${rem(16)} ${rem(16)} 0 ${rem(16)};
  div + div {
    margin-left: ${rem(12)};
  }
`;

export const FilterButtonsWrapper = styled.div`
  display: flex;
  flex-direction: row;
  padding: ${rem(24)} ${rem(16)} 0 ${rem(16)};
  div + div {
    margin-left: ${rem(12)};
  }
  div {
    flex-grow: 1;
  }
`;

export const CalendarButtonWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-left: ${rem(16)};
  margin-top: ${rem(24)};
  margin-bottom: ${rem(24)};
  margin-right: ${rem(16)};

  div:only-child {
    padding: 0 ${rem(24)} 0 ${rem(24)};
  }

  @media (min-width: 600px) {
    margin-left: 0;
    margin-right: 0;
  }
`;

export const IconView = styled.div`
  display: flex;
  justify-content: flex-end;
  align-items: center;
  height: 100%;
  padding: ${rem(16)};
`;

export const ClicableItem = styled.div`
  cursor: pointer;
`;

export const AccountBalanceWrapper = styled.div`
  display: flex;
  flex-direction: row;
  padding: 0 ${rem(8)} ${rem(24)} ${rem(0)};
`;

export const ErrorMessage = styled.div`
  display: flex;
  flex-direction: row;
  padding: ${rem(8)} ${rem(16)} 0 ${rem(16)};
  font-family: Lato;
  font-style: normal;
  font-weight: 500;
  font-size: ${rem(12)};
  line-height: 125%;
  letter-spacing: ${rem(0.45)};
  color: ${negative300};
`;

export const MobileStatementsWrapper = styled.div`
  max-width: ${rem(600)};
  margin: 0 auto;
`;

export const BtnWrapper = styled.div`
  flex-grow: 1;
  position: relative;
  flex-basis: 0;
`;
