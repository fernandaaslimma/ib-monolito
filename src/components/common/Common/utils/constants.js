export const PT_BR = "pt-BR";
export const EN_US = "en-US";
export const ZH_CN = "zh-CN";
export const SCREEN_LG = 1199;
export const SCREEN_MD = 992;
export const SCREEN_SM = 767;
export const SCREEN_XS = 449;
export const SCREEN_XXS = 319;
export const UP = "up";
export const DOWN = "down";
export const RIGHT = "right";
export const PAGINATION_TOTAL = "X-Total-Count";
export const DEFAULT_MOBILE_HEADER_HEIGHT = 60;
export const ASCENDING = "ASC";
export const DESCENDING = "DESC";
