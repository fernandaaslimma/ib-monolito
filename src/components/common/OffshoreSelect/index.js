import OffshoreSelect from "./OffshoreSelect";

export default OffshoreSelect;
