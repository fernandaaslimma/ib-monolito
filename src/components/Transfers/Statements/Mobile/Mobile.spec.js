import React from "react";
import { shallow } from "enzyme";

import Mobile from "./Mobile";
import DefaultContent from "../../../common/DefaultContent";
import { Icon, Button } from "react-bocombbm-components";

import store from "../../../../utils/store";
import { INDIVIDUAL } from "../../../../utils/constants";
import { Context } from "../../../common/OffshoreSelect/offshoreContext";

jest.mock("../../../../utils/store", () => ({
  getState: jest.fn()
}));

jest.mock("../../../../utils/i18n", () => ({
  getDateFieldPlaceholderByLocale: () => "MM/DD/YYY",
  getShortDateByLocale: () => "MM/DD/YYY",
  translate: id => id,
  getLanguage: () => "en-US",
  isZhCN: () => false
}));

const context = {
  currentCoin: "USD"
}

const futureEventsHistoryMock = [
  {
    availableAmount: 0,
    blockedAmount: 0,
    totalAmount: -3981.89,
    date: "2021-03-25",
    events: [
      {
        absAmount: 12.42,
        amount: -12.42,
        date: "2021-03-25",
        description: "TRANSFERENCIA ENTRE CONTAS DE MESMA TITULARIDADE",
        type: "Débito",
        settled: false,
        id: 12435901
      }
    ]
  },
  {
    availableAmount: 0,
    blockedAmount: 0,
    totalAmount: -3957.05,
    date: "2021-03-24",
    events: [
      {
        absAmount: 1912.13,
        amount: -1912.13,
        date: "2021-03-24",
        description: "TRANSFERENCIA ENTRE CONTAS DE MESMA TITULARIDADE",
        type: "Débito",
        settled: false,
        id: 12435889
      },
      {
        absAmount: 12.38,
        amount: -12.38,
        date: "2021-03-24",
        description: "TRANSFERENCIA ENTRE CONTAS DE DIFERENTES TITULARIDADES",
        type: "Débito",
        settled: false,
        id: 12435897
      },
      {
        absAmount: 120.41,
        amount: -120.41,
        date: "2021-03-24",
        description: "TRANSFERENCIA ENTRE CONTAS DE MESMA TITULARIDADE",
        type: "Débito",
        settled: false,
        id: 12435899
      }
    ]
  }
];

const history = {
  eventsSumAmount: "600",
  blockedBalance: 0,
  availableBalance: "400",
  totalBalance: "600",
  availableAmount: "41111",
  blockedAmount: 2,
  events: [
    {
      type: "type",
      description: "description",
      counterpart: "counterpart",
      value: "value"
    },
    {
      type: "CRÉDITO",
      description: "REPASSE COBRANÇA",
      counterpart: "BOCOM BBM S.A",
      value: "540.00"
    },
    {
      type: "CRÉDITO",
      description: "TRANSFERÊNCIA ENTRE CONTAS DE MESMA TITULARIDADE	",
      counterpart: "BOCOM BBM S.A",
      value: "-350.00"
    },
    {
      type: "CRÉDITO",
      description: "REPASSE COBRANÇA",
      counterpart: "BOCOM BBM S.A",
      value: "-570.00"
    },
    {
      type: "CRÉDITO",
      description: "OP: EMISSÃO DE TIT. CETIP",
      counterpart: "BOCOM BBM S.A",
      value: "-450.00"
    }
  ]
};

const props = {
  changeValuesVisibility: jest.fn(),
  getBalanceAndEventsHistory: jest.fn(),
  changeAccount: jest.fn(),
  getPendingOperationsNumber: jest.fn(),
  onFilter: jest.fn(),
  resetStates: jest.fn(),
  isGlobalMode: false,    
  originAccount: { accountNumber: "107 2 304020-3", blockedBalance: 0 },
  isEmptyEvents: false,
  accounts: [
    {
      accountNumber: "107 2 304020-3",
      blockedBalance: 0,
      document: "11111111111111"
    },
    {
      accountNumber: "107 2 304020-4",
      blockedBalance: 2000,
      document: "11111111111111"
    },
    { accountNumber: "107 2 304020-5", document: "11111111111111" }
  ],
  balanceHistoryParams: {
    range: {
      from: "12/01/2020",
      to: "12/02/2020"
    },
    activePage: 1,
    limit: 60,
    offset: 0,
    onlyDaysWithTransaction: true
  },
  balanceAndEventsHistory: [history],
  hideValues: false,
  filterButtonFill: { fillType: "period" },
  loading: false,
  futureEventsHistory: futureEventsHistoryMock
};

describe("Statements mobile", () => {
  let setState, useStateSpy, useEffect;
  const mockUseEffect = () => {
    useEffect.mockImplementation(f => f());
  };
  beforeEach(() => {
    setState = jest.fn();
    useStateSpy = jest.spyOn(React, "useState");
    useStateSpy.mockImplementation(init => [init, setState]);

    useEffect = jest.spyOn(React, "useEffect");
    mockUseEffect();

    store.getState.mockImplementation(() => ({
      userInfo: {
        tenants: [INDIVIDUAL]
      }
    }));
  });

  it("should match snapshot", () => {
    expect(shallow(
      <Context.Provider value={context}>
        <Mobile {...props} />
      </Context.Provider>
    )).toMatchSnapshot();
  });

  it("should match snapshot", () => {
    const newProps = {
      ...props,
      filterButtonFill: {
        fillType: "range",
        fillValue: { from: "24/04/2019", to: "24/04/2019" }
      }
    };
    expect(shallow(
      <Context.Provider value={context}>
        <Mobile {...newProps} />
      </Context.Provider>
    )).toMatchSnapshot();
  });

  it("should match snapshot", () => {
    const newProps = {
      ...props,
      filterButtonFill: {
        fillType: "range",
        fillValue: { from: "mock", to: "" }
      }
    };
    expect(shallow(
      <Context.Provider value={context}>
        <Mobile {...newProps} />
      </Context.Provider>
    )).toMatchSnapshot();
  });

  it.skip("should click to change content to radio", () => {
    const shallowComponent = shallow(
      <Context.Provider value={context}>
        <Mobile {...props} />
      </Context.Provider>
    );

    shallowComponent
      .find(Icon)
      .at(0)
      .simulate("click");

    expect(setState).toHaveBeenCalledWith("radioContent");
    expect(setState).toHaveBeenCalledWith(true);
  });

  it.skip("should click to change content to filter", () => {
    const shallowComponent = shallow(
      <Context.Provider value={context}>
        <Mobile {...props} />
      </Context.Provider>
    );

    shallowComponent
      .find(Button)
      .at(0)
      .simulate("click");

    expect(setState).toHaveBeenCalledWith("filterContent");
    expect(setState).toHaveBeenCalledWith(true);
  });

  it.skip("should click to change content to filter with fillType as range", () => {
    const newProps = {
      ...props,
      filterButtonFill: {
        fillType: "range",
        fillValue: { from: "mock", to: "mock" }
      }
    };

    const shallowComponent = shallow(
      <Context.Provider value={context}>
        <Mobile {...newProps} />
      </Context.Provider>
    );

    shallowComponent
      .find(Button)
      .at(0)
      .simulate("click");

    expect(setState).toHaveBeenCalledWith("filterContent");
    expect(setState).toHaveBeenCalledWith(true);
  });

  it("should match snapshot with only one account", () => {
    props.accounts = [
      {
        accountNumber: "107 2 304020-3",
        blockedBalance: 0,
        document: "11111111111111"
      }
    ];
    expect(shallow(
      <Context.Provider value={context}>
        <Mobile {...props} />
      </Context.Provider>
    )).toMatchSnapshot();
  });

  it("should match snapshot without statements", () => {
    props.accounts = [
      {
        accountNumber: "107 2 304020-3",
        blockedBalance: 0,
        document: "11111111111111"
      }
    ];
    props.balanceAndEventsHistory = [];
    props.isEmptyEvents = true;
    expect(shallow(
      <Context.Provider value={context}>
        <Mobile {...props} />
      </Context.Provider>
    )).toMatchSnapshot();
  });

  it.skip("should click to change account", () => {
    props.accounts = [
      {
        accountNumber: "107 2 304020-3",
        blockedBalance: 0,
        document: "11111111111111"
      }
    ];
    props.balanceAndEventsHistory = [];
    props.isEmptyEvents = true;
    props.isGlobalMode = false;
    props.filterButtonFill = undefined;

    const shallowComponent = shallow(
      <Context.Provider value={context}>
        <Mobile {...props} />
      </Context.Provider>
    );

    expect(shallowComponent.find(DefaultContent).length).toEqual(1);
  });  
});
