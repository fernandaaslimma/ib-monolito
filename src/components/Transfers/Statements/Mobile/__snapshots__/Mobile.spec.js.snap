// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statements mobile should match snapshot 1`] = `
<Mobile
  accounts={
    Array [
      Object {
        "accountNumber": "107 2 304020-3",
        "blockedBalance": 0,
        "document": "11111111111111",
      },
      Object {
        "accountNumber": "107 2 304020-4",
        "blockedBalance": 2000,
        "document": "11111111111111",
      },
      Object {
        "accountNumber": "107 2 304020-5",
        "document": "11111111111111",
      },
    ]
  }
  balanceAndEventsHistory={
    Array [
      Object {
        "availableAmount": "41111",
        "availableBalance": "400",
        "blockedAmount": 2,
        "blockedBalance": 0,
        "events": Array [
          Object {
            "counterpart": "counterpart",
            "description": "description",
            "type": "type",
            "value": "value",
          },
          Object {
            "counterpart": "BOCOM BBM S.A",
            "description": "REPASSE COBRANÇA",
            "type": "CRÉDITO",
            "value": "540.00",
          },
          Object {
            "counterpart": "BOCOM BBM S.A",
            "description": "TRANSFERÊNCIA ENTRE CONTAS DE MESMA TITULARIDADE	",
            "type": "CRÉDITO",
            "value": "-350.00",
          },
          Object {
            "counterpart": "BOCOM BBM S.A",
            "description": "REPASSE COBRANÇA",
            "type": "CRÉDITO",
            "value": "-570.00",
          },
          Object {
            "counterpart": "BOCOM BBM S.A",
            "description": "OP: EMISSÃO DE TIT. CETIP",
            "type": "CRÉDITO",
            "value": "-450.00",
          },
        ],
        "eventsSumAmount": "600",
        "totalBalance": "600",
      },
    ]
  }
  balanceHistoryParams={
    Object {
      "activePage": 1,
      "limit": 60,
      "offset": 0,
      "onlyDaysWithTransaction": true,
      "range": Object {
        "from": "12/01/2020",
        "to": "12/02/2020",
      },
    }
  }
  changeAccount={[MockFunction]}
  changeValuesVisibility={[MockFunction]}
  filterButtonFill={
    Object {
      "fillType": "period",
    }
  }
  futureEventsHistory={
    Array [
      Object {
        "availableAmount": 0,
        "blockedAmount": 0,
        "date": "2021-03-25",
        "events": Array [
          Object {
            "absAmount": 12.42,
            "amount": -12.42,
            "date": "2021-03-25",
            "description": "TRANSFERENCIA ENTRE CONTAS DE MESMA TITULARIDADE",
            "id": 12435901,
            "settled": false,
            "type": "Débito",
          },
        ],
        "totalAmount": -3981.89,
      },
      Object {
        "availableAmount": 0,
        "blockedAmount": 0,
        "date": "2021-03-24",
        "events": Array [
          Object {
            "absAmount": 1912.13,
            "amount": -1912.13,
            "date": "2021-03-24",
            "description": "TRANSFERENCIA ENTRE CONTAS DE MESMA TITULARIDADE",
            "id": 12435889,
            "settled": false,
            "type": "Débito",
          },
          Object {
            "absAmount": 12.38,
            "amount": -12.38,
            "date": "2021-03-24",
            "description": "TRANSFERENCIA ENTRE CONTAS DE DIFERENTES TITULARIDADES",
            "id": 12435897,
            "settled": false,
            "type": "Débito",
          },
          Object {
            "absAmount": 120.41,
            "amount": -120.41,
            "date": "2021-03-24",
            "description": "TRANSFERENCIA ENTRE CONTAS DE MESMA TITULARIDADE",
            "id": 12435899,
            "settled": false,
            "type": "Débito",
          },
        ],
        "totalAmount": -3957.05,
      },
    ]
  }
  getBalanceAndEventsHistory={[MockFunction]}
  getPendingOperationsNumber={[MockFunction]}
  hideValues={false}
  isEmptyEvents={false}
  isGlobalMode={false}
  loading={false}
  onFilter={[MockFunction]}
  originAccount={
    Object {
      "accountNumber": "107 2 304020-3",
      "blockedBalance": 0,
    }
  }
  resetStates={[MockFunction]}
/>
`;

exports[`Statements mobile should match snapshot 2`] = `
<Mobile
  accounts={
    Array [
      Object {
        "accountNumber": "107 2 304020-3",
        "blockedBalance": 0,
        "document": "11111111111111",
      },
      Object {
        "accountNumber": "107 2 304020-4",
        "blockedBalance": 2000,
        "document": "11111111111111",
      },
      Object {
        "accountNumber": "107 2 304020-5",
        "document": "11111111111111",
      },
    ]
  }
  balanceAndEventsHistory={
    Array [
      Object {
        "availableAmount": "41111",
        "availableBalance": "400",
        "blockedAmount": 2,
        "blockedBalance": 0,
        "events": Array [
          Object {
            "counterpart": "counterpart",
            "description": "description",
            "type": "type",
            "value": "value",
          },
          Object {
            "counterpart": "BOCOM BBM S.A",
            "description": "REPASSE COBRANÇA",
            "type": "CRÉDITO",
            "value": "540.00",
          },
          Object {
            "counterpart": "BOCOM BBM S.A",
            "description": "TRANSFERÊNCIA ENTRE CONTAS DE MESMA TITULARIDADE	",
            "type": "CRÉDITO",
            "value": "-350.00",
          },
          Object {
            "counterpart": "BOCOM BBM S.A",
            "description": "REPASSE COBRANÇA",
            "type": "CRÉDITO",
            "value": "-570.00",
          },
          Object {
            "counterpart": "BOCOM BBM S.A",
            "description": "OP: EMISSÃO DE TIT. CETIP",
            "type": "CRÉDITO",
            "value": "-450.00",
          },
        ],
        "eventsSumAmount": "600",
        "totalBalance": "600",
      },
    ]
  }
  balanceHistoryParams={
    Object {
      "activePage": 1,
      "limit": 60,
      "offset": 0,
      "onlyDaysWithTransaction": true,
      "range": Object {
        "from": "12/01/2020",
        "to": "12/02/2020",
      },
    }
  }
  changeAccount={[MockFunction]}
  changeValuesVisibility={[MockFunction]}
  filterButtonFill={
    Object {
      "fillType": "range",
      "fillValue": Object {
        "from": "24/04/2019",
        "to": "24/04/2019",
      },
    }
  }
  futureEventsHistory={
    Array [
      Object {
        "availableAmount": 0,
        "blockedAmount": 0,
        "date": "2021-03-25",
        "events": Array [
          Object {
            "absAmount": 12.42,
            "amount": -12.42,
            "date": "2021-03-25",
            "description": "TRANSFERENCIA ENTRE CONTAS DE MESMA TITULARIDADE",
            "id": 12435901,
            "settled": false,
            "type": "Débito",
          },
        ],
        "totalAmount": -3981.89,
      },
      Object {
        "availableAmount": 0,
        "blockedAmount": 0,
        "date": "2021-03-24",
        "events": Array [
          Object {
            "absAmount": 1912.13,
            "amount": -1912.13,
            "date": "2021-03-24",
            "description": "TRANSFERENCIA ENTRE CONTAS DE MESMA TITULARIDADE",
            "id": 12435889,
            "settled": false,
            "type": "Débito",
          },
          Object {
            "absAmount": 12.38,
            "amount": -12.38,
            "date": "2021-03-24",
            "description": "TRANSFERENCIA ENTRE CONTAS DE DIFERENTES TITULARIDADES",
            "id": 12435897,
            "settled": false,
            "type": "Débito",
          },
          Object {
            "absAmount": 120.41,
            "amount": -120.41,
            "date": "2021-03-24",
            "description": "TRANSFERENCIA ENTRE CONTAS DE MESMA TITULARIDADE",
            "id": 12435899,
            "settled": false,
            "type": "Débito",
          },
        ],
        "totalAmount": -3957.05,
      },
    ]
  }
  getBalanceAndEventsHistory={[MockFunction]}
  getPendingOperationsNumber={[MockFunction]}
  hideValues={false}
  isEmptyEvents={false}
  isGlobalMode={false}
  loading={false}
  onFilter={[MockFunction]}
  originAccount={
    Object {
      "accountNumber": "107 2 304020-3",
      "blockedBalance": 0,
    }
  }
  resetStates={[MockFunction]}
/>
`;

exports[`Statements mobile should match snapshot 3`] = `
<Mobile
  accounts={
    Array [
      Object {
        "accountNumber": "107 2 304020-3",
        "blockedBalance": 0,
        "document": "11111111111111",
      },
      Object {
        "accountNumber": "107 2 304020-4",
        "blockedBalance": 2000,
        "document": "11111111111111",
      },
      Object {
        "accountNumber": "107 2 304020-5",
        "document": "11111111111111",
      },
    ]
  }
  balanceAndEventsHistory={
    Array [
      Object {
        "availableAmount": "41111",
        "availableBalance": "400",
        "blockedAmount": 2,
        "blockedBalance": 0,
        "events": Array [
          Object {
            "counterpart": "counterpart",
            "description": "description",
            "type": "type",
            "value": "value",
          },
          Object {
            "counterpart": "BOCOM BBM S.A",
            "description": "REPASSE COBRANÇA",
            "type": "CRÉDITO",
            "value": "540.00",
          },
          Object {
            "counterpart": "BOCOM BBM S.A",
            "description": "TRANSFERÊNCIA ENTRE CONTAS DE MESMA TITULARIDADE	",
            "type": "CRÉDITO",
            "value": "-350.00",
          },
          Object {
            "counterpart": "BOCOM BBM S.A",
            "description": "REPASSE COBRANÇA",
            "type": "CRÉDITO",
            "value": "-570.00",
          },
          Object {
            "counterpart": "BOCOM BBM S.A",
            "description": "OP: EMISSÃO DE TIT. CETIP",
            "type": "CRÉDITO",
            "value": "-450.00",
          },
        ],
        "eventsSumAmount": "600",
        "totalBalance": "600",
      },
    ]
  }
  balanceHistoryParams={
    Object {
      "activePage": 1,
      "limit": 60,
      "offset": 0,
      "onlyDaysWithTransaction": true,
      "range": Object {
        "from": "12/01/2020",
        "to": "12/02/2020",
      },
    }
  }
  changeAccount={[MockFunction]}
  changeValuesVisibility={[MockFunction]}
  filterButtonFill={
    Object {
      "fillType": "range",
      "fillValue": Object {
        "from": "mock",
        "to": "",
      },
    }
  }
  futureEventsHistory={
    Array [
      Object {
        "availableAmount": 0,
        "blockedAmount": 0,
        "date": "2021-03-25",
        "events": Array [
          Object {
            "absAmount": 12.42,
            "amount": -12.42,
            "date": "2021-03-25",
            "description": "TRANSFERENCIA ENTRE CONTAS DE MESMA TITULARIDADE",
            "id": 12435901,
            "settled": false,
            "type": "Débito",
          },
        ],
        "totalAmount": -3981.89,
      },
      Object {
        "availableAmount": 0,
        "blockedAmount": 0,
        "date": "2021-03-24",
        "events": Array [
          Object {
            "absAmount": 1912.13,
            "amount": -1912.13,
            "date": "2021-03-24",
            "description": "TRANSFERENCIA ENTRE CONTAS DE MESMA TITULARIDADE",
            "id": 12435889,
            "settled": false,
            "type": "Débito",
          },
          Object {
            "absAmount": 12.38,
            "amount": -12.38,
            "date": "2021-03-24",
            "description": "TRANSFERENCIA ENTRE CONTAS DE DIFERENTES TITULARIDADES",
            "id": 12435897,
            "settled": false,
            "type": "Débito",
          },
          Object {
            "absAmount": 120.41,
            "amount": -120.41,
            "date": "2021-03-24",
            "description": "TRANSFERENCIA ENTRE CONTAS DE MESMA TITULARIDADE",
            "id": 12435899,
            "settled": false,
            "type": "Débito",
          },
        ],
        "totalAmount": -3957.05,
      },
    ]
  }
  getBalanceAndEventsHistory={[MockFunction]}
  getPendingOperationsNumber={[MockFunction]}
  hideValues={false}
  isEmptyEvents={false}
  isGlobalMode={false}
  loading={false}
  onFilter={[MockFunction]}
  originAccount={
    Object {
      "accountNumber": "107 2 304020-3",
      "blockedBalance": 0,
    }
  }
  resetStates={[MockFunction]}
/>
`;

exports[`Statements mobile should match snapshot with only one account 1`] = `
<Mobile
  accounts={
    Array [
      Object {
        "accountNumber": "107 2 304020-3",
        "blockedBalance": 0,
        "document": "11111111111111",
      },
    ]
  }
  balanceAndEventsHistory={
    Array [
      Object {
        "availableAmount": "41111",
        "availableBalance": "400",
        "blockedAmount": 2,
        "blockedBalance": 0,
        "events": Array [
          Object {
            "counterpart": "counterpart",
            "description": "description",
            "type": "type",
            "value": "value",
          },
          Object {
            "counterpart": "BOCOM BBM S.A",
            "description": "REPASSE COBRANÇA",
            "type": "CRÉDITO",
            "value": "540.00",
          },
          Object {
            "counterpart": "BOCOM BBM S.A",
            "description": "TRANSFERÊNCIA ENTRE CONTAS DE MESMA TITULARIDADE	",
            "type": "CRÉDITO",
            "value": "-350.00",
          },
          Object {
            "counterpart": "BOCOM BBM S.A",
            "description": "REPASSE COBRANÇA",
            "type": "CRÉDITO",
            "value": "-570.00",
          },
          Object {
            "counterpart": "BOCOM BBM S.A",
            "description": "OP: EMISSÃO DE TIT. CETIP",
            "type": "CRÉDITO",
            "value": "-450.00",
          },
        ],
        "eventsSumAmount": "600",
        "totalBalance": "600",
      },
    ]
  }
  balanceHistoryParams={
    Object {
      "activePage": 1,
      "limit": 60,
      "offset": 0,
      "onlyDaysWithTransaction": true,
      "range": Object {
        "from": "12/01/2020",
        "to": "12/02/2020",
      },
    }
  }
  changeAccount={[MockFunction]}
  changeValuesVisibility={[MockFunction]}
  filterButtonFill={
    Object {
      "fillType": "period",
    }
  }
  futureEventsHistory={
    Array [
      Object {
        "availableAmount": 0,
        "blockedAmount": 0,
        "date": "2021-03-25",
        "events": Array [
          Object {
            "absAmount": 12.42,
            "amount": -12.42,
            "date": "2021-03-25",
            "description": "TRANSFERENCIA ENTRE CONTAS DE MESMA TITULARIDADE",
            "id": 12435901,
            "settled": false,
            "type": "Débito",
          },
        ],
        "totalAmount": -3981.89,
      },
      Object {
        "availableAmount": 0,
        "blockedAmount": 0,
        "date": "2021-03-24",
        "events": Array [
          Object {
            "absAmount": 1912.13,
            "amount": -1912.13,
            "date": "2021-03-24",
            "description": "TRANSFERENCIA ENTRE CONTAS DE MESMA TITULARIDADE",
            "id": 12435889,
            "settled": false,
            "type": "Débito",
          },
          Object {
            "absAmount": 12.38,
            "amount": -12.38,
            "date": "2021-03-24",
            "description": "TRANSFERENCIA ENTRE CONTAS DE DIFERENTES TITULARIDADES",
            "id": 12435897,
            "settled": false,
            "type": "Débito",
          },
          Object {
            "absAmount": 120.41,
            "amount": -120.41,
            "date": "2021-03-24",
            "description": "TRANSFERENCIA ENTRE CONTAS DE MESMA TITULARIDADE",
            "id": 12435899,
            "settled": false,
            "type": "Débito",
          },
        ],
        "totalAmount": -3957.05,
      },
    ]
  }
  getBalanceAndEventsHistory={[MockFunction]}
  getPendingOperationsNumber={[MockFunction]}
  hideValues={false}
  isEmptyEvents={false}
  isGlobalMode={false}
  loading={false}
  onFilter={[MockFunction]}
  originAccount={
    Object {
      "accountNumber": "107 2 304020-3",
      "blockedBalance": 0,
    }
  }
  resetStates={[MockFunction]}
/>
`;

exports[`Statements mobile should match snapshot without statements 1`] = `
<Mobile
  accounts={
    Array [
      Object {
        "accountNumber": "107 2 304020-3",
        "blockedBalance": 0,
        "document": "11111111111111",
      },
    ]
  }
  balanceAndEventsHistory={Array []}
  balanceHistoryParams={
    Object {
      "activePage": 1,
      "limit": 60,
      "offset": 0,
      "onlyDaysWithTransaction": true,
      "range": Object {
        "from": "12/01/2020",
        "to": "12/02/2020",
      },
    }
  }
  changeAccount={[MockFunction]}
  changeValuesVisibility={[MockFunction]}
  filterButtonFill={
    Object {
      "fillType": "period",
    }
  }
  futureEventsHistory={
    Array [
      Object {
        "availableAmount": 0,
        "blockedAmount": 0,
        "date": "2021-03-25",
        "events": Array [
          Object {
            "absAmount": 12.42,
            "amount": -12.42,
            "date": "2021-03-25",
            "description": "TRANSFERENCIA ENTRE CONTAS DE MESMA TITULARIDADE",
            "id": 12435901,
            "settled": false,
            "type": "Débito",
          },
        ],
        "totalAmount": -3981.89,
      },
      Object {
        "availableAmount": 0,
        "blockedAmount": 0,
        "date": "2021-03-24",
        "events": Array [
          Object {
            "absAmount": 1912.13,
            "amount": -1912.13,
            "date": "2021-03-24",
            "description": "TRANSFERENCIA ENTRE CONTAS DE MESMA TITULARIDADE",
            "id": 12435889,
            "settled": false,
            "type": "Débito",
          },
          Object {
            "absAmount": 12.38,
            "amount": -12.38,
            "date": "2021-03-24",
            "description": "TRANSFERENCIA ENTRE CONTAS DE DIFERENTES TITULARIDADES",
            "id": 12435897,
            "settled": false,
            "type": "Débito",
          },
          Object {
            "absAmount": 120.41,
            "amount": -120.41,
            "date": "2021-03-24",
            "description": "TRANSFERENCIA ENTRE CONTAS DE MESMA TITULARIDADE",
            "id": 12435899,
            "settled": false,
            "type": "Débito",
          },
        ],
        "totalAmount": -3957.05,
      },
    ]
  }
  getBalanceAndEventsHistory={[MockFunction]}
  getPendingOperationsNumber={[MockFunction]}
  hideValues={false}
  isEmptyEvents={true}
  isGlobalMode={false}
  loading={false}
  onFilter={[MockFunction]}
  originAccount={
    Object {
      "accountNumber": "107 2 304020-3",
      "blockedBalance": 0,
    }
  }
  resetStates={[MockFunction]}
/>
`;
